core/src/test/java/com/sequenceiq/cloudbreak/service/eventbus/CloudResourceRetrievalServiceTest.java

```diff
@@ -0,0 +1,83 @@
+package com.sequenceiq.cloudbreak.service.eventbus;
+
+import static com.sequenceiq.common.api.type.CommonStatus.REQUESTED;
+import static com.sequenceiq.common.api.type.ResourceType.AZURE_MANAGED_IMAGE;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.Optional;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.core.convert.ConversionService;
+
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.domain.Resource;
+import com.sequenceiq.cloudbreak.service.resource.ResourceService;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CloudResourceRetrievalServiceTest {
+
+    private static final String RESOURCE_REFERENCE = "resource-reference";
+
+    private static final String RESOURCE = "resource1";
+
+    @InjectMocks
+    private CloudResourceRetrievalService underTest;
+
+    @Mock
+    private ConversionService conversionService;
+
+    @Mock
+    private ResourceService resourceService;
+
+    @Test
+    public void testGetByResourceReferenceShouldReturnCloudResource() {
+        Resource resource = createResource();
+        CloudResource cloudResource = createCloudResource();
+
+        when(resourceService.findByResourceReferenceAndStatusAndType(RESOURCE_REFERENCE, REQUESTED, AZURE_MANAGED_IMAGE)).thenReturn(Optional.of(resource));
+        when(conversionService.convert(resource, CloudResource.class)).thenReturn(cloudResource);
+
+        Optional<CloudResource> actual = underTest.findByResourceReferenceAndStatusAndType(RESOURCE_REFERENCE, REQUESTED, AZURE_MANAGED_IMAGE);
+
+        assertTrue(actual.isPresent());
+        assertEquals(cloudResource, actual.get());
+        verify(resourceService).findByResourceReferenceAndStatusAndType(RESOURCE_REFERENCE, REQUESTED, AZURE_MANAGED_IMAGE);
+        verify(conversionService).convert(resource, CloudResource.class);
+    }
+
+    @Test
+    public void testGetByResourceReferenceShouldReturnCloudResourceShouldReturnEmptyList() {
```

Comment:
The test name is strange, I guess you wanted testGetByResourceReferenceShouldReturnEmptyList